spring-core/src/main/java/org/springframework/util/ReflectionUtils.java

```diff
@@ -486,6 +488,18 @@ private static Method[] getDeclaredMethods(Class<?> clazz, boolean defensive) {
 		return (result.length == 0 || !defensive) ? result : result.clone();
 	}
 
+	private static int methodDeclaringClassHierarchySorter(Method method1, Method method2) {
+		Class<?> clazz1 = method1.getDeclaringClass();
+		Class<?> clazz2 = method2.getDeclaringClass();
+		if (clazz1 == clazz2) {
```

Comment:
For methods in the array returned from `java.lang.Class.getDeclaredMethods()`, the declaring class is **always** the same class. Thus, there is no point in comparing the declaring classes.